### PR TITLE
[WIP] Update instruction for digital alliance to move to new standard environment.

### PR DESCRIPTION
**Pull request by blaisb**
### Description

Our installation instructions for the digital alliance cluster were dated because they relied on the previous 2020 Standard environment. I have updated them to work on the 2023 environment. I find they are simpler now, which is good.

@lpsaavedra maybe we should add the AVX-512 instructions by default?

Pull request related list:
- [X] Labels are applied
- [X] There are at least 2 reviewers (or 1 if small feature) excluding the responsible for the merge
- [X] If this PR closes an issue or is related to a project, it is linked in the "Projects" or "Development" section
- [X] If any future works is planed, an issue is opened
- [X] The PR description is cleaned and ready for merge

### CODE REVIEW

**Comment by lpsaavedra** [comment]
I will test them right now using Niagara... I will add the AVX-512 instructions in an additional note since it is only for Niagara

**Review by oguevremont — APPROVED** [review]
I recompiled everything using these instructions and it works

File: doc/source/installation/digital_alliance.rst [hunk 1]
```diff
@@ -33,24 +33,20 @@ On Niagara, Beluga, Narval, Graham or Cedar
...
-  module load openmpi/4.0.3
-  module load trilinos/13.4.1
+  module load StdEnv/2023
+  module load trilinos/15.1.1
   module load parmetis/4.0.3
-  module load p4est/2.2
-  module load opencascade/7.5.2
-  module load muparser/2.3.2
+  module load muparser/2.3.4
+  module lost p4est/2.8.6
```
Comment (oguevremont) [inline doc/source/installation/digital_alliance.rst:47]:
```suggestion
  module load p4est/2.8.6
```

File: doc/source/installation/digital_alliance.rst [hunk 2]
```diff
@@ -163,16 +157,11 @@ In the nano terminal, copy-paste (with ``Ctrl+Shift+V``):
...
-  module load trilinos/13.4.1
+  module load StdEnv/2023
+  module load trilinos/15.1.1
   module load parmetis/4.0.3
-  module load p4est/2.2
-  module load opencascade/7.5.2
-  module load muparser/2.3.2
-
+  module load muparser/2.3.4
+  module lost p4est/2.8.6
```
Comment (oguevremont) [inline doc/source/installation/digital_alliance.rst:164]:
```suggestion
  module load p4est/2.8.6
```

**Review by oguevremont — CHANGES_REQUESTED** [review]
I initially missed this

**Review by lpsaavedra — CHANGES_REQUESTED** [review]
I am having issues with this version of Trilinos and Kokkos. I got some warnings when installing deal.II and then Lethe with prototypes on. I will look into this.

**Review by lpsaavedra — APPROVED** [review]
Fixed the compilaiton of the prototypes in #1260. The Kokkos warnings are still present but that should not be an issue at the moment since we only use Kokkos for a prototype... we will have to test this more

**Review by oguevremont — APPROVED** [review]

File: doc/source/installation/digital_alliance.rst [hunk 3]
```diff
@@ -64,18 +60,20 @@ We can compile ``dealii`` in the ``$HOME/dealii/build`` folder, by defining the
...
+  cmake ../dealii -DDEAL_II_WITH_MPI=ON -DDEAL_II_WITH_TRILINOS=ON   -DTRILINOS_DIR=$EBROOTTRILINOS  -DDEAL_II_WITH_P4EST=ON -DCMAKE_INSTALL_PREFIX=$HOME/dealii/inst/ -DDEAL_II_WITH_METIS=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=../inst/ -DCMAKE_SKIP_INSTALL_RPATH=ON -DDEAL_II_COMPONENT_EXAMPLES=OFF -G Ninja
```
[the remark this answers is not in the document]
Comment (blaisb) [reply doc/source/installation/digital_alliance.rst:63]: We actually don't need it i think, i'm not sure i am still testing this.

**Comment by blaisb** [comment]
> I will test them right now using Niagara... I will add the AVX-512 instructions in an additional note since it is only for Niagara

AVX would also work on Beluga as far as I know.

**Comment by lpsaavedra** [comment]
> AVX would also work on Beluga as far as I know.
Not sure... they do not mention anything in the Beluga page. However, in the Narval page it says: "Narval does not however support the AVX512instruction set, in contrast to the nodes of Béluga and Niagara as well as certain nodes of Cedar and Graham."... so I am guessing you are right. Do you want me to test the AVX instructions on Beluga?

**Comment by blaisb** [comment]
> > AVX would also work on Beluga as far as I know.
> 
> Not sure... they do not mention anything in the Beluga page. However, in the Narval page it says: "Narval does not however support the AVX512instruction set, in contrast to the nodes of Béluga and Niagara as well as certain nodes of Cedar and Graham."... so I am guessing you are right. Do you want me to test the AVX instructions on Beluga?

Nah let's leave it as is for this PR.
:)

**Pull request merged**